core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/AiServicesProcessor.java

```diff
@@ -230,6 +231,29 @@ public void findDeclarativeServices(CombinedIndexBuildItem indexBuildItem,
                 }
             }
 
+            DotName retrievalAugmentorSupplierClassDotName = null;
+            AnnotationValue retrievalAugmentorSupplierValue = instance.value("retrievalAugmentor");
+            if (retrievalAugmentorSupplierValue != null) {
+                retrievalAugmentorSupplierClassDotName = retrievalAugmentorSupplierValue.asClass().name();
+                if (NO_RETRIEVAL_AUGMENTOR.equals(retrieverClassDotName)) {
+                    retrievalAugmentorSupplierClassDotName = null;
+                } else {
+                    BuiltinScope declaredScope = BuiltinScope
+                            .from(index.getClassByName(retrievalAugmentorSupplierClassDotName));
+                    if (declaredScope == null) {
+                        throw new IllegalConfigurationException(
+                                "Unable to determine the CDI scope of " + retrievalAugmentorSupplierClassDotName
+                                        + ". Is it a CDI bean?");
+                    }
```

Comment:
Wouldn't it be better to use the pattern we have for other types like `BeanChatMemoryProviderSupplier` in order to not have to force users to create a bean of type `Supplier<RetrievalAugmentor>`, but what they actually need which is a `RetrievalAugmentor`?